### ISSUE TITLE
Shuffling the test dataset leads to a dramatic performance decrease.

### ISSUE THREAD

**Issue by downw** [QEpiphany/MSGAT#1]
I noticed that you shuffled the training dataset but did not shuffle the test dataset. When I reproduce the results, the HR@20 on Diginetica is 68.39. However, when I shuffle the test dataset, the HR@20 on Diginetica drop to 47.35. Could you explain this discrepancy to me?
Thank you.

**Comment by QEpiphany** [comment]
Thank you very much for your question. I think this phenomenon occurs because of an important component in MSGAT - the intent neighbor collaboration module. This module ideally searches for session representations with similar intentions from the entire dataset for selective enhancement. However, since this process is too computationally intensive and our laboratory's GPU cannot handle it, here we only search for sessions with similar intentions in the same batch. Since the dataset is enhanced by data segmentation, there are many sequences with similar items in the same batch. When the test set is not shuffled, the number of actual similar sessions in the batch is greater than the number of similar sessions set by the hyperparameters, which can provide enough effective information to enhance the current session representation. When the test set is shuffled, the number of similar sessions contained in the same batch is very small, and these low-similarity sessions cannot provide enough effective information for enhancement of the current session representation.

**Comment by downw** [comment]
Thank you!
However, this issue might lead to a similar problem in the code of this paper: https://github.com/YuhanZhen/WSDM23-DGNNs--for-Session-based-Recommendation/issues/2, specifically data leakage.

**Comment by QEpiphany** [comment]
Thank you for your tip, but I think this is not a data leakage problem, because the method of shuffling the train set but not shuffling the test set is used in many papers, such as SRGNN. Our code in the data processing part follows the original code of SRGNN and has not been modified. Unlike DGNN, neither the train set nor the test set of DGNN has been shuffled, so their model will learn the data distribution during training, resulting in data leakage.
In addition, for the test set, the reason why shuffle is generally not performed is that the test phase is mainly to evaluate the performance of the model on unseen data, and what is pursued is a true reflection of the model's generalization ability. Keeping the original order of the test set can help us more accurately understand the performance of the model in various situations that may be encountered in practical applications.

**Comment by QEpiphany** [comment]
I'm sorry for this phenomenon in MSGAT, but this performance change is more likely to reflect the model's insufficient generalization ability or sensitivity to data distribution or increased evaluation noise, rather than a data leak.

**Comment by downw** [comment]
Thanks for your response
